application/classes/Ushahidi/Repository/Tag.php

```diff
@@ -116,7 +116,8 @@ public function create(Entity $entity)
 	public function update(Entity $entity)
 	{
 		$tag = $entity->getChanged();
-
+		// removing children before saving tag
+		unset($tag['children']);
```

Comment:
Ah immutable has no effect, because the previous value was 'empty' so it just sets it :/